### PR TITLE
Changing md5 to sha256

### PR DESCRIPTION
**Pull request by rishi-anand**
@jzhoucliqr Please review

### CODE REVIEW

File: pkg/util/util/util.go [hunk 1]
```diff
@@ -45,9 +45,9 @@ func RandIDWithLen(idLen int) (id string, err error) {
 
 func GetAuthKey(token string, timestamp int64) (key string) {
 	token = token + fmt.Sprintf("%d", timestamp)
-	md5Ctx := md5.New()
-	md5Ctx.Write([]byte(token))
-	data := md5Ctx.Sum(nil)
+	sha256Ctx := sha256.New()
```
Comment (jzhoucliqr) [inline pkg/util/util/util.go:48]: what's the timestamp here?what's the reason you said that old client won't work? becaue if the timestamp is current timestamp, then to me this function is probably just called by one party to generate a key, not by two parties to generate the same key. But I'm just guessing.

File: build/frpc/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.16 as builder
+FROM golang:1.18.1 as builder
```
Comment (jzhoucliqr) [inline build/frpc/Dockerfile]: shall we use 1.19?

Comment (rishi-anand) [reply build/frpc/Dockerfile]: Sure Jun

File: pkg/util/util/util.go (hunk 1 above)
Comment (rishi-anand) [reply pkg/util/util/util.go:48]: @jzhoucliqr Client is not able to authenticate with server with these changes. Need to deep dive into the code to find out how it is being used, but it feels like we will need to host 2 proxy server for a while.

```
(base) rishi@Rishis-MacBook-Pro ~ $ k logs -f spectro-proxy-7f875568d8-gr7c7
Defaulted container "frpc" out of: frpc, frpc-init (init)
2023/02/06 01:19:33 [E] [service.go:340] token in login doesn't match token from configuration
2023/02/06 01:19:33 [W] [service.go:128] login to server failed: token in login doesn't match token from configuration
token in login doesn't match token from configuration
(base) rishi@Rishis-MacBook-Pro ~ $
```

File: build/frpc/Dockerfile (hunk 2 above)
Comment (rishi-anand) [reply build/frpc/Dockerfile]: Fixed in https://github.com/spectrocloud/frp/pull/2/commits/7d947f628fcabb3d771f5827f78069b1482fe7d2

**Review by jzhoucliqr — APPROVED** [review]

**Pull request merged**